src/Logic/calculate.js

```diff
@@ -0,0 +1,77 @@
+import operate from './operate';
+
+const calculate = (data, name) => {
+  const { total, next, operation } = data;
+
+  if (name === '+/-') {
+    return {
+      next: `${next * (-1)}`,
+      operation,
+      total,
+    };
+  }
+
+  if (name === '%') {
+    return {
+      total: `${operate(next, total, operation)}`,
+      next: null,
+      operation: '%',
+    };
+  }
+
+  if (name === '/') {
+    return {
+      total: `${operate(next, total, operation)}`,
+      operation: '/',
+      next: '',
+    };
+  }
+
+  if (name === 'X') {
+    return {
+      total: `${operate(next, total, operation)}`,
+      next: '0',
+      operation: 'X',
+    };
+  }
+
+  if (name === '+') {
+    return {
```

Comment:
### [Optional]
It is a good practice to use `switch case` once our condition is more than 3;
kindly check out this article 👉  [JS switch vs if-else](https://medium.com/the-daily-standup/javascript-switch-statement-vs-if-else-if-ee3111793fa0